### PR TITLE
introduce testscript, fix return bug, add map keys sorting, non color mode bug fix

### PR DESCRIPTION
**Pull request by ldemailly**
- [x] Introduce testscript
- [x] Failing test
- [x] Fixed bug reset color code being printed in non color mode
- [x] Sort object keys by type then content
- [x] Adding test of sample.gr's output
- [x] Fix bug #46

### CODE REVIEW

File: eval/eval.go [hunk 1]
```diff
@@ -304,7 +304,7 @@ func (s *State) applyFunction(fn object.Object, args []object.Object) object.Obj
 	}
 	curState := s.env
 	s.env = nenv
-	res := s.evalInternal(function.Body)
+	res := s.Eval(function.Body) // Need to have the return value unwrapped. Bug #46
```
Comment (ccoVeille) [inline eval/eval.go:307]: Either it's a todo


```suggestion
	res := s.Eval(function.Body) // TODONeed to have the return value unwrapped. Bug #46
```

Either is was fixed and I would expect to do not mention "bug"

```suggestion
	res := s.Eval(function.Body) // Need to have the return value unwrapped. See #46
```

Comment (ldemailly) [reply eval/eval.go:307]: it _fixes_ Bug #46, I can add Fixes #46, there is no todo it's the fix

**Review by ldemailly — COMMENTED** [review]
.

File: eval/eval.go (hunk 1 above)
Comment (ldemailly) [reply eval/eval.go:307]: fixed in https://github.com/grol-io/grol/pull/53/commits/4097534d4e86fab40035cb38c8cf087d598da88d

**Comment by ccoVeille** [comment]
LGTM, even if I'm reviewing once merged

**Pull request merged**